controllers/pipeline/pipeline_adapter.go

```diff
@@ -195,6 +196,21 @@ func (a *Adapter) EnsureApplicationSnapshotPassedAllTests() (results.OperationRe
 	return results.ContinueProcessing()
 }
 
+// EnsureStatusReported will ensure that integration PipelineRun status is reported to the git provider
+// which (indirectly) triggered its execution.
+func (a *Adapter) EnsureStatusReported() (results.OperationResult, error) {
+	if !tekton.IsIntegrationPipelineRun(a.pipelineRun) {
```

Comment:
I think we'll also need to check if the pipeline was run for the `pull_request` event type by checking the [PipelineAsCodeEventTypeLabel](pipelinesascode.tekton.dev/event-type) on the pipelineRun.

Reply:
This check is now performed by each `Reporter` implementation (e.g. https://github.com/redhat-appstudio/integration-service/pull/61/files#diff-e3efbea6f0e41cd39d3da695764ffb088284311c7a59a8c5b53a27fffe33d8f1R167). This should make it flexible should we ever wish to also react to PipelineRun events on a provider-by-provider basis.

Reply:
Cool, that makes sense